src/SDKs/Azure.Core/data-plane/Azure.Core/Pipeline/Policies/TelemetryPolicy.cs

```diff
@@ -11,35 +11,56 @@ namespace Azure.Core.Pipeline.Policies
 {
     public class TelemetryPolicy : HttpPipelinePolicy
     {
-        private readonly HttpHeader _header;
+        private readonly Assembly _clientAssembly;
 
-        public TelemetryPolicy(Assembly clientAssembly, string applicationId)
+        private HttpHeader _header;
+
+        private string _applicationId;
+
+        public string ApplicationId
+        {
+            get => _applicationId;
+            set
+            {
+                _applicationId = value;
+                InitializeHeader();
+            }
+        }
+
+        public TelemetryPolicy(Assembly clientAssembly)
```

Comment:
You added the attributes to make it automatic. Now, there is a ctor where users have to pass the assembly. This seems like unnecessary indirection and more complicate ctor.

Reply:
I don't think anyone except `ClientOptions` type would create an instance of this. Assembly is used to avoid hardcoding both the name and the version of the client which is harder to track manually.